src/main/java/org/broadinstitute/hellbender/engine/spark/GATKSparkTool.java

```diff
@@ -176,12 +179,28 @@ public SAMFileHeader getHeaderForReads() {
     }
 
     /**
-     * Loads the reads into a {@link JavaRDD} using the intervals specified. If no intervals
-     * were specified, returns all the reads (both mapped and unmapped).
+     * Loads the reads into a {@link JavaRDD} using the intervals specified, and filters them using
+     * the filter returned by {@link #makeReadFilter}.
      *
-     * @return all reads from our reads input(s) as a {@link JavaRDD}, bounded by intervals if specified.
+     * If no intervals were specified, returns all the reads (both mapped and unmapped).
+     *
+     * @return all reads from our reads input(s) as a {@link JavaRDD}, bounded by intervals if specified,
+     *         and filtered using the filter from {@link #makeReadFilter}.
      */
     public JavaRDD<GATKRead> getReads() {
+        final ReadFilter filter = makeReadFilter();
+        return getUnfilteredReads().filter(read -> filter.apply(read));
```

Comment:
tiny micro optimization that probably doesn't matter at all becauser of JIT, but you should prefer `filter.test` over `filter.apply`